File: Algorithms/Strings/The Love-Letter Mystery.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

class Result
{

    /*
     * Complete the 'theLoveLetterMystery' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts STRING s as parameter.
     */

    public static int theLoveLetterMystery(string s)
    {
        int count = 0;
        for (int i = 0; i < s.Length / 2; i++)
        {
            count += Math.Abs(s[i] - s[s.Length - 1 - i]);
        }
        return count;
    }
}

class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        int q = Convert.ToInt32(Console.ReadLine().Trim());

        for (int qItr = 0; qItr < q; qItr++)
        {
            string s = Console.ReadLine();

            int result = Result.theLoveLetterMystery(s);

            textWriter.WriteLine(result);
        }

        textWriter.Flush();
        textWriter.Close();
    }
}

File: Algorithms/Strings/The Love-Letter Mystery.php
<?php

/*
 * Complete the 'theLoveLetterMystery' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts STRING s as parameter.
 */

function theLoveLetterMystery($s) {
    # Write your code here
    $sum = 0;
    for ($i = 0; $i < strlen($s) / 2; $i++) {
        $sum += abs(ord($s[$i]) - ord($s[strlen($s) - 1 - $i]));
    }
    return (int)$sum;
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$q = intval(trim(fgets(STDIN)));

for ($q_itr = 0; $q_itr < $q; $q_itr++) {
    $s = rtrim(fgets(STDIN), "\r\n");

    $result = theLoveLetterMystery($s);

    fwrite($fptr, $result . "\n");
}

fclose($fptr);
